components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -183,6 +178,10 @@ tobeDeleted claims (claims came from federated idp as null). If there is a match
                 userClaims.remove(FrameworkConstants.PASSWORD);
                 boolean userWorkflowEngaged = false;
                 try {
+                    // This thread local is set to skip the password pattern validation if it is a generated one.
+                    if (!isUserProvidedPassword) {
+                        UserCoreUtil.setSkipPasswordPatternValidationThreadLocal(true);
```

Comment:
Is this threadlocal used in both userstore layer as well in identity governance connector?

Reply:
Yes. This is used in both places to skip the validation.